comp110-worksheet-6/OxoBoard.cs

```diff
@@ -4,73 +4,196 @@
+        public Mark GetSquare(int x, int y)
+        {
+            return board[x, y];
+        }
+        // If the specified square is currently empty, fill it with mark and return true.
+        // If the square is not empty, leave it as-is and return False.
+
+            //Check if the coordinates, x and y, don't surpass the boundaries.
+
+        public bool SetSquare(int x, int y, Mark mark)
+        {
+            int counter;
+
+            if ((x >= 0 && x < 3) && (y >= 0 && y <3))
+            {
+                if (board[x, y] == Mark.None)
+                {
+                    board[x, y] = mark;
+                    return true;
+                }
+            }
+            return false;
+
+        }
+
+        //counter++;
+
+
+        // If there are still empty squares on the board, return false.
+        // If there are no empty squares, return true.
+
+        //Iterate through each square and check if it is empty. If it it is, the board is not full, therefore return false. Otherwise, return true after finish iterating through each one of them.
+        public bool IsBoardFull()
+        {
+
+
+            for (int i = 0; i < width; i++)
+                for (int j = 0; j < height; j++)
+                    if (board[i, j] == Mark.None)
+                        return false;
+            return true;
+
+
+
+        }
+
+        // If a player has three in a row, return Mark.O or Mark.X depending on which player.
+        // Otherwise, return Mark.None.
+        public Mark GetWinner()
+        {
+            //Check for vertical marks + making sure the square that is in a row is not empty
+            for (int i = 0; i < 3; i++)
+            {
+                if (board[i, 0] == board[i, 1] && board[i, 1] == board[i, 2] && GetSquare(i, 0) != Mark.None)
+                    return GetSquare(i, 0);
+
+            }
+
+            //Check for horizontal marks making sure the square that is in a row is not empty
+            for (int j = 0; j < 3; j++)
+            {
+                if (board[0, j] == board[1, j] && board[1, j] == board[2, j] && GetSquare(0, j) != Mark.None)
+                    return GetSquare(0, j);
+
+            }
+
+            //Check for diagonal marks making sure the square that is in a row is not empty
+            if ((board[0, 0] == board[1, 1] && board[1, 1] == board[2, 2] || board[0, 2] == board[1, 1] && board[1, 1] == board[2, 0]) && GetSquare(1, 1) != Mark.None)
+                return GetSquare(1, 1);
+
+            /*
+            if (board[0, 2] == board[1, 1] && board[1, 1] == board[2, 0])
+                return GetSquare(0, 2);
+            */
+
+
+            return Mark.None;
+        }
+
+
+
+
+        // Strech goal attempts below for nxn board + 'k' in a row.
+
+        
+        public bool isHorizontal(int i, int j)
+        {
+            int counter = 0;
+            for (int y = 0; y < height; y++)
+                if (board[i, j] == board[i, y])
+                    counter++;
```

Comment:
You could say `else return false` here -- as soon as you encounter a square not equal to the first square, you know you do not have a line